index-scheduler/src/lib.rs

```diff
@@ -378,37 +421,93 @@ impl IndexScheduler {
     pub fn get_task_ids(&self, query: &Query) -> Result<RoaringBitmap> {
         let rtxn = self.env.read_txn()?;
 
-        // This is the list of all the tasks.
-        let mut tasks = self.all_task_ids(&rtxn)?;
+        let ProcessingTasks { started_at: started_at_processing, processing: tasks_processing } =
+            self.processing_tasks.read().unwrap().clone();
 
-        if let Some(uids) = &query.uid {
-            tasks &= RoaringBitmap::from_iter(uids);
-        }
+        let mut tasks = self.all_task_ids(&rtxn)?;
 
         if let Some(status) = &query.status {
+            let mut include_processing_tasks = false;
             let mut status_tasks = RoaringBitmap::new();
             for status in status {
-                status_tasks |= self.get_status(&rtxn, *status)?;
+                match status {
+                    // special case for Processing tasks
+                    Status::Processing => {
+                        include_processing_tasks = true;
+                        status_tasks |= &tasks_processing;
+                    }
+                    status => status_tasks |= &self.get_status(&rtxn, *status)?,
+                };
+            }
+            if !include_processing_tasks {
+                tasks -= &tasks_processing;
```

Comment:
```suggestion
                    Status::Processing => status_tasks |= &tasks_processing,
                    status => status_tasks |= &self.get_status(&rtxn, *status)?,
                }
            }
            if status.contains(&Status::Processing) {
                tasks -= &tasks_processing;
```

You can remove this `include_processing_tasks` boolean and simplify a bit, here.